python/ray/tests/test_advanced.py

```diff
@@ -603,9 +604,44 @@ def sum(numbers):
                 assert not inlined
         return result
 
-    results = f.remote()
-    g_ref = sum.remote(results)
-    assert ray.get(g_ref) == 10
+    assert ray.get(sum.remote()) == 10
+
+
+def test_task_arguments_inline_bytes_limit(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(
+        num_cpus=1,
+        resources={"pin_head": 1},
+        _system_config={
+            "max_direct_call_object_size": 100 * 1024,
+            # if task_rpc_inlined_bytes_limit is greater than
+            # max_grpc_message_size, this test fails.
+            "task_rpc_inlined_bytes_limit": 18 * 1024,
+            "max_grpc_message_size": 20 * 1024,
+            "put_small_object_in_memory_store": True,
+        },
+    )
+    cluster.add_node(num_cpus=1, resources={"pin_worker": 1})
+    ray.init(address=cluster.address)
+
+    @ray.remote(resources={"pin_worker": 1})
+    def foo(ref1, ref2, ref3):
+        return ref1 == ref2 + ref3
+
+    @ray.remote(resources={"pin_head": 1})
+    def bar():
+        # if the refs are inlined, the test fails.
```

Comment:
if we pass by ref and the referenced objects are stored locally (instead of in plasma), the current PR were not able to protect against it. However I can't find the code where we inline the referenced object into RPCs..  cc @rkooo567 @ericl any pointer?

Reply:
Pass by ref should never inline I believe.